### PR TITLE
Add support for Rails 5.2.3

### PR DESCRIPTION
**Pull request by mriddle**
Upon upgrading to Rails 5.2.3 and using v0.11.0 of this gem I received exceptions when a forked process attempted to communicate to the database. 

I could reproduce the behaviour with `config.cache_classes = config.eager_load = true`  set in my `config/environment/development.rb` and executing  `rails runner "Process.fork { ActiveRecord::Base.connected? }"`

Rails has modified the way it handles database connections in
 * rails/rails#31173 
 * rails/rails#28057 
 * rails/rails#31221

This patch adds supports to this gem for Rails 5.2.3 with the following 
 * Don't auto-create connection if nil on methods 5916c96
 * Clear entry from PoolProxy when Rails discards a ConnectionPool 460b4d1

### CODE REVIEW

**Comment by mriddle** [comment]
These changes have also been verified against a Rails 5.1 application.

File: test/test_arhp.rb [hunk 1]
```diff
@@ -13,6 +13,23 @@ def teardown
...
   end
 
+  def test_process_forking_with_connections
+    # Ensure we have a connection already
+    assert_equal(true, ActiveRecord::Base.connected?)
+
+    # Verify that when we fork, the process doesn't crash
+    pid = Process.fork do
+      if ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR == 2
+        assert_equal(false, ActiveRecord::Base.connected?) # New to Rails 5.2
```
Comment (bquorning) [inline test/test_arhp.rb]: Did this test fail before the code change was applied?

Comment (mriddle) [reply test/test_arhp.rb]: Yep!

Comment (mriddle) [reply test/test_arhp.rb]: I verified the fix in 3 different ways
 * Stashing the changes to `pool_proxy.rb` and running this unit test
 * Booting an application, forking a process and making a request to the DB
 * Deploying a Rails 5.2 application with these changes and running smoke tests

**Comment by bquorning** [comment]
I’d like to have an extra set of eyes on this PR – @grosser ?

File: Readme.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ You need a local user called 'john-doe'.
     GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX  ON *.* TO 'john-doe'@'localhost';
     FLUSH PRIVILEGES;
 
+With mysql running locally, run
+
+    BUNDLE_GEMFILE=gemfiles/rails5.2.gemfile bundle exec rake test
+
+ Or
+
+    BUNDLE_GEMFILE=gemfiles/rails5.2.gemfile ruby -Ilib:test test/test_arhp.rb --seed 19911 --verbose
```
[the remark this answers is not in the document]
Comment (mriddle) [reply Readme.md]: You're right. It's not necessary, I'll remove it. 🙇

File: lib/active_record_host_pool/pool_proxy.rb [hunk 3]
```diff
@@ -84,6 +84,34 @@ def clear_reloadable_connections!
       _clear_connection_proxy_cache
     end
 
+    def release_connection(owner_thread = Thread.current)
+      p = _connection_pool(false)
```
[the remark this answers is not in the document]
Comment (mriddle) [reply lib/active_record_host_pool/pool_proxy.rb]: It does, but I'd prefer to leave the 3 methods as-is. It's easier to grok at a glance IMO.

**Comment by mriddle** [comment]
Thanks for the feedback and quick reviews! 🙏

**Pull request merged**